src/register-stories.js

```diff
@@ -0,0 +1,63 @@
+const Vue = require('vue')
+
+const capitalize = ([first, ...rest], lowerRest = false) =>
+  first.toUpperCase() + (lowerRest ? rest.join('').toLowerCase() : rest.join(''));
+
+const camelCase = str => {
+  let s =
+    str &&
+    str
+      .match(/[A-Z]{2,}(?=[A-Z][a-z]+[0-9]*|\b)|[A-Z]?[a-z]+[0-9]*|[A-Z]|[0-9]+/g)
+      .map(x => x.slice(0, 1).toUpperCase() + x.slice(1).toLowerCase())
+      .join('');
+  return s.slice(0, 1).toLowerCase() + s.slice(1);
+};
+
+function registerStories (req, fileName, sb, plugins) {
+  const {
+    action,
+    withNotes,
+    text,
+    boolean,
+    number,
+    color,
+    object,
+    array,
+    select,
+    date
+  } = plugins
+  const componentConfig = req(fileName)
+  const componentName = capitalize(
+    camelCase(
+      fileName
+        .replace(/^\.\/_/, '')
+        .replace(/\.\w+$/, '')
+    )
+  )
+
+  const stories = componentConfig.__stories || componentConfig.default.__stories
+  if (!stories) return
+  stories.forEach(story => {
+    console.log(story)
+    let addFunc
+    let baseFunc = () => {
+      let data = story.knobs ? eval(`(${story.knobs})`) : {}
+      return {
+        data () {
+          return data
+        },
+        template: story.template,
+        methods: eval(`(${story.methods})`)
+      }
+    }
+
+    story.notes
+    ? addFunc = withNotes(story.notes)(baseFunc)
+    : addFunc = baseFunc
+
+    sb(story.name, module).add(story.name, addFunc)
+    Vue.default.component(componentName, componentConfig.default || componentConfig)
```

Comment:
I don't think I'm familiar with the `Vue.default` property. Should this just be `Vue.component`?

Reply:
If I'm not mistaken, I did this because I'm requiring Vue (not importing it) at the top of the file. There wasn't a component prop on the required Vue object when I tested locally.

Reply:
Ah, I see! Then I'd probably update that to `const Vue = require('vue').default`, then here, you won't have to worry about the fact that `Vue` isn't actually the Vue object.